Tests/RemoteDumpTest.php

```diff
@@ -171,12 +236,160 @@ public function failIfLaravelSanctumIsDisabledAndNoHtaccessIsDefined()
+    }
+
+    /**
+     * @test
+     */
+    public function failIfNoDatabaseConnectionIsSet()
+    {
+        Config::set('database.connections', null);
+
+        $result = $this->protector->configure();
+
+        $this->assertFalse($result);
+    }
+
+    /**
+     * @test
+     */
+    public function canGetConfigValueForKey()
+    {
+        Config::set('protector.baseDirectory', __FUNCTION__);
+
+        $method = $this->getAccessibleReflectionMethod('getConfigValueForKey');
+        $result = $method->invoke($this->protector, 'baseDirectory');
+
+        $this->assertEquals(__FUNCTION__, $result);
+    }
+
+    /**
+     * @test
+     */
+    public function canResolveBaseDirectoryFromClosure()
+    {
+        $method = $this->getAccessibleReflectionMethod('getConfigValueForKey');
+
+        Config::set('protector.baseDirectory', fn() => 'null');
+        $result = $method->invoke($this->protector, 'baseDirectory');
+
+        $this->assertEquals('null', $result);
+    }
+
+    /**
+     * @test
+     */
+    public function validateUsersPrivateKeyName()
+    {
+        $this->protector->setPrivateKeyName(__FUNCTION__);
+        $privateKeyName = $this->protector->getPrivateKeyName();
+
+        $this->assertEquals(__FUNCTION__, $privateKeyName);
+    }
+
+    /**
+     * @test
+     */
+    public function failDecryptingOnInvalidString()
+    {
+        $this->expectException(InvalidConfigurationException::class);
+        $this->protector->decryptString(base64_encode(__FUNCTION__));
+    }
+
+    /**
+     * @test
+     */
+    public function canCreateFilename()
+    {
+        $fileName = $this->protector->createFilename();
+
+        $this->assertIsString($fileName);
+    }
+
+    /**
+     * @test
+     */
+    public function canGetAuthToken()
+    {
+        $methodGet = $this->getAccessibleReflectionMethod('getAuthToken');
+        $authToken = $methodGet->invoke($this->protector);
+
+        $this->assertEquals(env('PROTECTOR_AUTH_TOKEN'), $authToken);
+    }
+
+    /**
+     * @test
+     */
+    public function setAuthToken()
+    {
+        $methodSet = $this->getAccessibleReflectionMethod('setAuthToken');
+        $methodGet = $this->getAccessibleReflectionMethod('getAuthToken');
+
+        $methodSet->invoke($this->protector, __FUNCTION__);
+        $authToken = $methodGet->invoke($this->protector);
+
+        $this->assertEquals(__FUNCTION__, $authToken);
+    }
+
+    /**
+     * @test
+     */
+    public function setAuthTokenKeyName()
+    {
+        $methodSet = $this->getAccessibleReflectionMethod('setAuthTokenKeyName');
+        $methodGet = $this->getAccessibleReflectionMethod('getAuthTokenKeyName');
+
+        $methodSet->invoke($this->protector, __FUNCTION__);
+        $authTokenKeyName = $methodGet->invoke($this->protector);
+
+        $this->assertEquals(__FUNCTION__, $authTokenKeyName);
+    }
+
+    /**
+     * @test
+     */
+    public function canGetServerUrl()
```

Comment:
this test is redundant due to the following test

Reply:
Done.